app/code/Magento/ConfigurableProductGraphQl/Model/Options/Collection.php

```diff
@@ -123,11 +133,31 @@ private function fetch() : array
             $this->attributeMap[$productId][$attribute->getId()]['id'] = $attribute->getId();
             $this->attributeMap[$productId][$attribute->getId()]['attribute_code']
                 = $attribute->getProductAttribute()->getAttributeCode();
-            $this->attributeMap[$productId][$attribute->getId()]['values'] = $attributeData['options'];
+            $this->attributeMap[$productId][$attribute->getId()]['values'] = $this->getAllowedOptions($attributeData);
             $this->attributeMap[$productId][$attribute->getId()]['label']
                 = $attribute->getProductAttribute()->getStoreLabel();
         }
 
         return $this->attributeMap;
     }
+
+    /**
+     * Returns List of Allowed Options
+     *
+     * @param $attributeData
+     * @return array
+     */
+    protected function getAllowedOptions($attributeData) {
+        $options = [];
+
+        foreach ($attributeData['options'] as $keyOption => $valOption) {
+            $configurableProduct = $this->productRepository->get($valOption['sku']);
```

Comment:
I afraid this approach can kill performance of the endpoint.
Maybe we should add plugin for GraphQL area which will remove disabled options from options select?
Maybe this will be right place to do so: \Magento\ConfigurableProduct\Model\ResourceModel\Attribute\OptionSelectBuilder::getSelect
@paliarush @akaplya we need your advice there.